### ISSUE TITLE
@PrimaryKey is not handled correctly when using insert()

### ISSUE THREAD

**Issue by optisamit** [realm/realm-java#3212]
#### Steps & Code to Reproduce

```java

public class PrimaryKeyActivity extends Activity {
    private static final String TAG = PrimaryKeyActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final Car car = new Car();
        car.wheels = new RealmList<>(new Wheel(), new Wheel());

        final Realm realm = Realm.getDefaultInstance();
        Log.d(TAG, "Number of wheels with id 1 before inserting: " + realm.where(Wheel.class).equalTo("id", 1).count());
        realm.beginTransaction();
        realm.insert(car);
        realm.commitTransaction();
        Log.d(TAG, "Number of wheels with id 1 after inserting: " + realm.where(Wheel.class).equalTo("id", 1).count());
        realm.close();
    }
}

public class Car extends RealmObject {
    @PrimaryKey
    private long id = 42;
    public RealmList<Wheel> wheels;
}

public class Wheel extends RealmObject {
    @PrimaryKey
    private long id = 1;
}

```

#### Expected log output:

>D/PrimaryKeyActivity: Number of wheels with id 1 before inserting: 0
>D/PrimaryKeyActivity: Number of wheels with id 1 after inserting: 1

#### Actiual log output:

>D/PrimaryKeyActivity: Number of wheels with id 1 before inserting: 0
>D/PrimaryKeyActivity: Number of wheels with id 1 after inserting: **2**

As you can see, we have two objects with the same primary key.
Note that `insertOrUpdate()` and `copyToRealm()` work correctly.

#### Version of Realm and tooling
Realm version(s): 1.1.0

Android Studio version: 2.1.2

Which Android version and device: Android 4.2.2 on Samsung GT-I9105

**Comment by Zhuinden** [comment]
From the docs:

https://realm.io/docs/java/latest/api/io/realm/Realm.html#insert-io.realm.RealmModel-


> insert
> public void insert(RealmModel object)
> Insert an unmanaged RealmObject. This is generally faster than copyToRealm(RealmModel) since it doesn't return the inserted elements, and performs minimum allocations and checks. After being inserted any changes to the original object will not be persisted.
> Please note:
> 
> - We don't check if the provided objects are already managed or not, so inserting a managed object will duplicate it
> - We don't create (nor return) a managed RealmObject for each element
> - Copying an object will copy all field values. Any unset field in the object and child objects will be set to their default value if not provided
>
> If you want these checks and the managed RealmObject returned, use copyToRealm(RealmModel), otherwise if you have a large number of object this method is generally faster.
> 
> Parameters:
> object - RealmObjects to insert.
> See Also:
> copyToRealm(RealmModel)

**We don't check if the provided objects are already managed or not, so inserting a managed object will duplicate it**

**Comment by optisamit** [comment]
It is also noteworthy that I'm not inserting a managed object directly: It is `Wheel` object that is duplicated, not `Car`. So, if I want to insert an object that has two references to another object, insert doesn't work at all. This should either be fixed or be made obvious

**Comment by stk1m1** [comment]
Hi @optisamit,

Thanks for the detailed report. It is reproducible and clearly needed to be fixed.

**Comment by cmelchior** [comment]
Fixed by #3239
